### PR TITLE
CORE-54 type checking tests for dangling nodes

### PR DESCRIPTION
**Pull request by cottand**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -27,6 +27,7 @@ module.exports = {
   overrides: [
     {
       files: [
+        '**/*.ts',
```
Comment (lancelotblanchard) [inline .eslintrc.js:30]: Why is this necessary?

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,8 @@ pnpm-debug.log*
 *.njsproj
 *.sln
 *.sw?
+
+src/**/*.js
```
Comment (lancelotblanchard) [inline .gitignore:25]: Why?

File: src/app/ir/istateToGraph.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ function toGraphNode(inode: INodeState): ModelNode {
 }
 
 export default function istateToGraph(istate: IState): Graph {
+  console.log(`JSON of istate\n${JSON.stringify(istate)}`);
```
Comment (lancelotblanchard) [inline src/app/ir/istateToGraph.ts:27]: I believe this is a debug log that we might want to remove?

**Review by lancelotblanchard — DISMISSED** [review]
Looks really good to me, congrats! Just a few things but feel free to merge whenever! Did you create a Frontend Ticket to reflect these errors visually?

**Review by tinasandu — DISMISSED** [review]

**Review by lancelotblanchard — APPROVED** [review]
🎉

File: .gitignore (hunk 2 above)
Comment (cottand) [reply .gitignore:25]: in order to ignore intellij-generated js files
